pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -522,21 +533,28 @@ def apply_lightning(self, operations):
                 method = getattr(self._gpu_state, name, None)
                 wires = self.wires.indices(ops.wires)
 
-                if method is None:
+                if ops.name == "C(GlobalPhase)":
+                    controls = ops.control_wires
+                    control_values = ops.control_values
+                    param = ops.base.parameters[0]
+                    matrix = global_phase_diagonal(param, self.wires, controls, control_values)
+                    self._gpu_state.apply(name, wires, False, [], matrix)
```

Comment:
The `global_phase_diagonal` says that it returns only the diagonal of the operator.
Are we only sending the diagonal to be applied? I got confused here.

Reply:
Yes, the diagonal is passed via the `matrix` argument down to `applyOperation` which uses a special routine when the name of the operation is `C(GlobalPhase)`. I used that hack to get something working quick in L-GPU/Kokkos. The proper way to do things will be to add controlled-kernels as in L-Qubit, after which we can delete `global_phase_diagonal` and the associated logic.

Reply:
OK